src/components/shared/MarkerTooltipContents.js

```diff
@@ -34,6 +47,50 @@ function getMarkerDetails(data: MarkerPayload): React$Element<*> | null {
           </div>
         );
       }
+      case 'GCMinor': {
+        if (data.nursery && data.nursery.reason) {
```

Comment:
For the nursery things. I will first change how Firefox produces those markers and then update the flow types to more tightly resemble that.  So I'll revise that in a different PR later. I experimented with flow and so far I'm happy, but I did find some situations it could not handle.

My question about the valid data from Firefox is party to understand how types change JavaScript from SpiderMonkey's perspective.  For example, if the flow types say that the nursery field is ALWAYS present and never null, then is it possible to give it data a null field and get a nastier null pointer dereference crash? How much does flow change the "compiled" JavaScript (possibly this changes when asm.js is involved).

For this work I will add comments to the relevant parts of FIrefox asking people to update the type definitions in perf.html if they change the data it can add to markers.

Thank you.